android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/DateTimeExtension.kt

```diff
@@ -147,5 +147,27 @@ fun calculateAge(date: Date, context: Context, localDateNow: LocalDate = LocalDa
   }.trim()
 }
 
+fun calculateAgeFromDays(ageInDays: Int, context: Context): String {
+  val period = Period.ofDays(ageInDays)
+  val years = period.days / 365
+  val months = (period.days % 365) / 30
+  val weeks = ((period.days % 365) % 30) / 7
+  val days = ((period.days % 365) % 30) % 7
```

Comment:
oh time, it's so complex.

1. on average there are 365.25 days in a year, so that'd be the number to divide by in general here
2. but since you are talking about days in a specific year, it'd be more accurate to use something like the below for all the specific years you're talking about
```java
Calendar cal = Calendar.getInstance();
int year = 2023;
int daysInYear = cal.getActualMaximum(Calendar.DAY_OF_YEAR);
```
3. but if you know the actual date of birth and can use that instead of an integer number of days, that would be the best way to do this
```java
Period period = Period.between(birthDate, currentDate);
System.out.println("Your age is " + period.getYears() + " years, " + period.getMonths() + " months, and " + period.getDays() + " days.");
```

h/t Bard

Reply:
@pld Resorted to using the patient's DoB to calculcate age on a given date.

We however encountered a challenge whereby the Patient resource/object is not accessible in a profile's views' rules. The same applies for rules defined in the main profile section - they are not accessible inside rules in the views section.

I will share a ticket to track this issue in FHIRCore separately.